tt_metal/hw/firmware/src/trisc.cc

```diff
@@ -107,11 +108,17 @@ int main(int argc, char *argv[]) {
         while (*trisc_run != RUN_SYNC_MSG_GO);
         DeviceZoneScopedMainN("TRISC-FW");
 
+        uint32_t kernel_config_base = mailboxes->launch.kernel_config.kernel_config_base;
+
 #if !defined(UCK_CHLKC_MATH)
-        setup_cb_read_write_interfaces(0, mailboxes->launch.kernel_config.max_cb_index, cb_init_read, cb_init_write);
+        uint32_t tt_l1_ptr *cb_l1_base = (uint32_t tt_l1_ptr *)(kernel_config_base +
+            mailboxes->launch.kernel_config.cb_offset);
+        setup_cb_read_write_interfaces(cb_l1_base, 0, mailboxes->launch.kernel_config.max_cb_index, cb_init_read, cb_init_write, cb_init_write);
+#if defined(UCK_CHLKC_UNPACK)
```

Comment:
Can you add a comment about why this delay was added with the issue link?

Reply:
yes, will do